Sources/Sentry/SentrySpan.m

```diff
@@ -84,10 +96,46 @@ - (instancetype)initWithContext:(SentrySpanContext *)context
         _spanId = context.spanId;
         _sampled = context.sampled;
         _origin = context.origin;
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        if (SentrySDK.options.enableContinuousProfiling) {
+            _profileSessionIDs = [NSMutableSet<NSString *> set];
+            if (SentryContinuousProfiler.isCurrentlyProfiling) {
+                [_profileSessionIDs
+                    addObject:SentryContinuousProfiler.currentProfilerID.sentryIdString];
+            } else {
+                [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
+                    addObserver:self
+                       selector:@selector(linkProfiler)
+                           name:kSentryNotificationContinuousProfileStarted];
+            }
+        }
+#endif // SENTRY_TARGET_PROFILING_SUPPORTED
     }
     return self;
 }
 
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+- (void)dealloc
+{
+    [self stopObservingContinuousProfiling];
+}
+
+- (void)linkProfiler
+{
+    [_profileSessionIDs addObject:SentryContinuousProfiler.currentProfilerID.sentryIdString];
+}
+
+- (void)stopObservingContinuousProfiling
+{
+    if (SentrySDK.options.enableContinuousProfiling) {
```

Comment:
`l`: Sorry, I that I didn't catch this in the first review. If somebody keeps a reference to the Options and changes `enableContinuousProfiling` while the span is running to `false`, we would never stop observing. That's unlikely, but maybe it's better to add property to the SentrySpan to know if we have to stop observing.

Reply:
Good idea 👍🏻